### PR TITLE
Changes for use of separate block devices for meta and data

### PR DESCRIPTION
**Pull request by agrover**

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -66,6 +66,8 @@ $(basename $0) options:
     -X        | xfstests git repo. Used by tests/xfstests.sh.
     -x        | xfstests git branch to checkout and track.
     -y        | xfstests ./check additional args
+    -z <file> | Specify the filesystem's meta data device path that contains
```
Comment (zabbo) [inline run-tests.sh]: -d and -z rubs me the wrong way.. maybe rename -D to -F for dumping ftrace on oops, and use -M for metadata and -D for data?  These single char options aren't looking like the best idea I've ever had.

File: run-tests.sh [hunk 2]
```diff
@@ -354,19 +369,21 @@ fi
 # mount concurrently so that a quorum is present to elect the leader and
 # start a server.
 #
-msg "mounting $T_NR_MOUNTS mounts on $T_DEVICE"
+msg "mounting $T_NR_MOUNTS mounts on $T_DATA_DEVICE"
```
Comment (zabbo) [inline run-tests.sh]: Maybe print both devices here?

File: run-tests.sh [hunk 3]
```diff
@@ -354,19 +369,21 @@ fi
...
 pids=""
 for i in $(seq 0 $((T_NR_MOUNTS - 1))); do
-	opts="-o server_addr=127.0.0.1"
 
-	dev=$(losetup --find --show $T_DEVICE)
-	test -b "$dev" || die "failed to create temp device $dev"
+	meta_dev=$(losetup --find --show $T_META_DEVICE)
+	test -b "$meta_dev" || die "failed to create temp device $meta_dev"
+	data_dev=$(losetup --find --show $T_DATA_DEVICE)
+	test -b "$meta_dev" || die "failed to create temp device $data_dev"
```
Comment (zabbo) [inline run-tests.sh]: meant to test data_dev..

File: run-tests.sh [hunk 4]
```diff
@@ -354,19 +369,21 @@ fi
...
 
 	dir="/mnt/test.$i"
 	test -d "$dir" || cmd mkdir -p "$dir"
 
-	msg "mounting $dev on $dir"
-	cmd mount -t scoutfs $opts "$dev" "$dir" &
+	msg "mounting $data_dev on $dir"
```
Comment (zabbo) [inline run-tests.sh]: print both again, probably?

File: run-tests.sh [hunk 5]
```diff
@@ -375,9 +392,13 @@ for i in $(seq 0 $((T_NR_MOUNTS - 1))); do
...
 	T_OS+="$opts "
 
-	eval T_B$i=$dev
-	T_B[$i]=$dev
-	T_BS+="$dev "
+	eval T_MB$i=$meta_dev
+	T_MB[$i]=$meta_dev
+	T_MBS+="$meta_dev "
+
+	eval T_B$i=$data_dev
```
Comment (zabbo) [inline run-tests.sh]: _DB to match _MB?

File: tests/setup-error-teardown.sh [hunk 6]
```diff
@@ -7,9 +7,9 @@
 
 t_require_commands sleep killall
 
-echo "== interrupt waiting mount with no options"
+echo "== interrupt waiting mount with minimal options"
```
Comment (zabbo) [inline tests/setup-error-teardown.sh]: Heh.  Might as well just remove everything after "waiting mount".

**Review by zabbo — CHANGES_REQUESTED** [review]
Definitely almost there, It's missing some mounts in funcs/fs.h, I guess the tests aren't all passing yet?

**Comment by agrover** [comment]
tests that don't require two mounts (or are xfstests) are all passing.

File: tests/setup-error-teardown.sh [hunk 7]
```diff
@@ -7,9 +7,9 @@
 
 t_require_commands sleep killall
 
-echo "== interrupt waiting mount with minimal options"
+echo "== interrupt waiting mount"
```
Comment (zabbo) [inline tests/setup-error-teardown.sh:10]: The golden output needs to be updated for this.

**Comment by zabbo** [comment]
> tests that don't require two mounts (or are xfstests) are all passing.

Nice!  And I think it'll be pretty easy to update xfstests.  It does mean that the single -e argument for the extra dev will want to turn into another pair of meta/data devices.  I'm not sure how best to work those into the options.. are single char options just a bad idea at this point?

**Comment by agrover** [comment]
I used `-e` and `-f` (why? It's after e :smiley: ). This is only temporary. I think we want to move -tests to long options, which will make this a nonissue.

**Comment by zabbo** [comment]
This was landed in another PR derived from this, with a few fixes added.

**Pull request closed without merging**